lib/BridgeManager.php

```diff
@@ -0,0 +1,399 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2020 Julien Veyssier <eneiluj@posteo.net>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Talk;
+
+use OCP\IConfig;
+use OCP\IDBConnection;
+use OCP\IL10N;
+use OCP\Files\IAppData;
+use OCP\Files\NotFoundException;
+use OCP\Files\SimpleFS\ISimpleFolder;
+use OCP\IURLGenerator;
+
+use OCA\Talk\Exceptions\ImpossibleToKillException;
+
+class BridgeManager {
+	public const EVENT_TOKEN_GENERATE = self::class . '::generateNewToken';
+
+	/** @var IDBConnection */
+	private $db;
+	/** @var IConfig */
+	private $config;
+	/** @var IAppData */
+	private $appData;
+	/** @var IL10N */
+	private $l;
+
+	public function __construct(IDBConnection $db,
+								IConfig $config,
+								IAppData $appData,
+								IURLGenerator $urlGenerator,
+								Manager $manager,
+								IL10N $l) {
+		$this->db = $db;
+		$this->config = $config;
+		$this->urlGenerator = $urlGenerator;
+		$this->appData = $appData;
+		$this->manager = $manager;
+		$this->l = $l;
+	}
+
+	public function getBridgeOfRoom(string $token): array {
+		$defaultParts = sprintf(
+			'{"enabled":false,"pid":0,"parts":[{"type":"nctalk","server":"","login":"","password":"","channel":"%s"}]}',
+			$token
+		);
+		$bridgeJSON = $this->config->getAppValue('spreed', 'bridge_' . $token, $defaultParts);
```

Comment:
Will this scale? App configs are loaded always and on each request. I think it should instead be stored in a dedicated database table.

Reply:
Maybe I'm missing something here but it seems to me that accessing a dedicated DB table will be the same as calling getAppValue as app values are stored in the DB.

We need the bridge information when loading the sidebar tab and when performing config editions. We can't do it less often.

Do you think of a better (and/or more efficient) way to store/get bridge configs?

Reply:
> Maybe I'm missing something here but it seems to me that accessing a dedicated DB table will be the same as calling getAppValue as app values are stored in the DB.

Well but appconfig is loaded also when you do something in deck, ping the root of files as a sync client, check for notifications, etc. Now if someone setups up like hundreds of bridges it will load hundreds of entries every time.

Reply:
Ok I'll put it in a dedicated table. Didn't think about the importance of keeping appconfig reasonably small. Thanks.